File: 프로그래머스/0/120866. 안전지대/안전지대.js
function solution(board) {
    let answer = 0;
    const mine = [[-1, 0], [1, 0], [0, 1], [0, -1], [-1, 1], [1, 1], [1, -1], [-1, -1]];
    
    for (let i = 0; i < board.length; i++) {
        for (let j = 0; j < board[i].length; j++) {
            if (board[i][j] == 1) {
                mine.map((v) => {
                    let [x, y] = v;
                    [x, y] = [x + i, y + j];
                    if (x >= 0 && x < board.length && y >= 0 && y < board[i].length && board[x][y] == 0) {
                        board[x][y] = 2;
                    }
                })
            }
        }
    }
    board.map(v1 => v1.map(v2 => v2 == 0 ? answer++ : 0));
    return answer;
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    answer = 0
    mine_deltas = [
        (-1, 0), (1, 0), (0, 1), (0, -1),
        (-1, 1), (1, 1), (1, -1), (-1, -1)
    ]
    rows = len(board)
    cols = len(board[0])
    for r in range(rows):
        for c in range(cols):
            if board[r][c] == 1:
                for dr, dc in mine_deltas:
                    nr, nc = r + dr, c + dc
                    if 0 <= nr < rows and 0 <= nc < cols:
                        if board[nr][nc] == 0:
                            board[nr][nc] = 2               
    for r in range(rows):
        for c in range(cols):
            if board[r][c] == 0:
                answer += 1    
    return answer